### PR TITLE
Core typing: utils6

### PR DESCRIPTION
**Pull request by gpotter2**
- typing of `scapy/utils.py`
- fix `in6_getRandomizedIfaceId` which was pretty broken.
- ~~requires https://github.com/secdev/scapy/pull/2908~~

### CODE REVIEW

File: scapy/utils6.py [hunk 1]
```diff
@@ -415,18 +448,17 @@ def in6_getRandomizedIfaceId(ifaceid, previous=None):
...
+        b_previous = inet_pton(socket.AF_INET6, "::" + previous)[8:]
+    s = inet_pton(socket.AF_INET6, "::" + ifaceid)[8:] + b_previous
     import hashlib
     s = hashlib.md5(s).digest()
     s1, s2 = s[:8], s[8:]
-    s1 = chb(orb(s1[0]) | 0x04) + s1[1:]
-    s1 = inet_ntop(socket.AF_INET6, b"\xff" * 8 + s1)[20:]
-    s2 = inet_ntop(socket.AF_INET6, b"\xff" * 8 + s2)[20:]
-    return (s1, s2)
+    s1 = chb(orb(s1[0]) & (~0x04)) + s1[1:]  # set bit 6 to 0
```
Comment (gpotter2) [inline scapy/utils6.py:458]: This was just broken (especially this line). I checked the spec

File: scapy/utils6.py [hunk 2]
```diff
@@ -884,22 +954,27 @@ def rec(n, li):
...
-        return reduce(operator.mul, ((y - x) for (x, y) in self.parsed), 1)
+        return reduce(operator.mul, ((y - x) for x, y in self.parsed), 1)
 
     def __str__(self):
+        # type: () -> str
         try:
             return next(self.__iter__())
         except (StopIteration, RuntimeError):
-            return None
+            return ""
```
Comment (gpotter2) [inline scapy/utils6.py:966]: This would raise an error on Python 3.

File: test/regression.uts [hunk 3]
```diff
@@ -2692,18 +2692,21 @@ in6_addrtomac("FE80::" + in6_mactoifaceid("FF:00:00:00:00:00", ulbit=0)) == 'ff:
 
 ########### RFC 3041 related function ###############################
 = Test in6_getRandomizedIfaceId
+
 import socket
 
-res=True
 for a in six.moves.range(10):
```
Comment (gpotter2) [inline test/regression.uts:2698]: This test tested nothing

**Review by gpotter2 — COMMENTED** [review]
Few bugs fixed

**Comment by gpotter2** [comment]
@guedou This should be ready for review

**Review by guedou — APPROVED** [review]

**Pull request merged**